msalautomationapp/src/main/java/com/microsoft/identity/client/msal/automationapp/sdk/MsalSdk.java

```diff
@@ -268,4 +290,49 @@ private IAccount getAccountForPolicyName(@NonNull final MultipleAccountPublicCli
         }
         return null;
     }
+
+    public String generateSHR(@NonNull MsalAuthTestParams authTestParams, TokenRequestTimeout tokenRequestTimeout) throws Throwable {
+        final IPublicClientApplication pca = setupPCA(
+                authTestParams.getActivity(),
+                authTestParams.getMsalConfigResourceId()
+        );
+
+        final IAccount account = getAccount(
+                authTestParams.getActivity(),
+                authTestParams.getMsalConfigResourceId(),
+                authTestParams.getLoginHint()
+        );
+
+        if (null == account) {
+            // User must first sign-in
+            return null;
+        }
+        final ResultFuture<String, Exception> future = new ResultFuture<>();
+        final IPublicClientApplication.SignedHttpRequestRequestCallback callback = new IPublicClientApplication.SignedHttpRequestRequestCallback() {
+            @Override
+            public void onTaskCompleted(String result) {
+                future.setResult(result);
+            }
+
+            @Override
+            public void onError(MsalException exception) {
+                future.setException(exception);
+            }
+
+        };
+
+        if (authTestParams.getAuthScheme() == AuthScheme.POP) {
```

Comment:
I think we don't necessarily need to check this here at all. We can just proceed to generateSignedHttpRequest regardless of it because just calling the `generateShr` method implies that scheme is PoP